centos.org/ansible/provision_duffy.yml

```diff
@@ -0,0 +1,33 @@
+---
+- hosts: localhost
+  tasks:
+    - name: 'Get a new node'
+      evgeni.duffy.session_request:
+        pool: metal-ec2-c5n-centos-8s-x86_64
+        quantity: 1
+      register: duffy_data
+
+    - name: 'Fail if no hosts returned'
+      fail:
+        msg: 'Could not get hosts from ci.centos.org'
+      when: duffy_data.nodes|length == 0
```

Comment:
Should `evgeni.duffy.session_request` fail on this?

Reply:
It might even already do, I am not sure ;)

Reply:
Doesn't look like it so I opened https://github.com/evgeni/evgeni.duffy/pull/3 so at least API errors are caught. If you then assume it returns a proper error if quantity could not be satified, I think this is fine.